未知原因的文章 404

### 系统信息

- 外部访问地址: https://www.anye.xyz
- 启动时间: 2024-08-10 16:49
- 版本: 2.18.0
- 构建时间: 2024-08-01 10:23
- Git Commit: 08aa7be
- Java: IBM Semeru Runtime Open Edition / 21.0.3+9-LTS
- 数据库: PostgreSQL / 16.3
- 操作系统: Linux / 5.15.0-107-generic
- 已激活主题: [Earth-Fixed 1.8.0](https://github.com/Anyexyz/theme-earth_fixed)
- 已启动插件:
  - [页面静态缓存 1.1.0](https://github.com/halo-sigs/plugin-page-cache)
  - [文章限制阅读 Pro 1.4.1](https://www.halo.run/store/apps/app-TzRqd)
  - [编辑器超链接卡片 1.0.3](https://github.com/halo-sigs/plugin-editor-hyperlink-card)
  - [Data Studio（数据工厂） 1.0.0-alpha.2](https://github.com/halo-sigs/plugin-data-studio)
  - 联系表单 1.0.2
  - [AI 助手 1.0.0](https://www.halo.run/store/apps/app-riNgb)
  - [静态网页服务 1.0.0-alpha.4](https://www.halo.run/store/apps/app-gFkMn)
  - [Docsme 1.0.0-alpha.8](https://www.halo.run/store/apps/app-yffxw)
  - [瞬间 1.7.0](https://github.com/halo-sigs/plugin-moments)
  - [文章导入导出 1.2.4](https://jhacker.cn/2023/halo-plugin-export2doc)
  - [文本绘图 1.3.0](https://github.com/halo-sigs/plugin-text-diagram)
  - [OAuth2 认证 1.3.0](https://github.com/halo-sigs/plugin-oauth2)
  - [TinyTale小程序设置 1.0.4](https://github.com/jiewenhuang/halo-plugin-TinyTale)
  - [评论组件 2.4.0](https://github.com/halo-dev/plugin-comment-widget)
  - [图库管理 1.4.0](https://github.com/halo-sigs/plugin-photos)
  - [链接管理 1.4.0](https://github.com/halo-sigs/plugin-links)
  - [通知我 1.1.0](https://github.com/monyuan/notify-me)
  - [Umami 1.4.0](https://github.com/halo-sigs/plugin-umami)
  - [增强备份（Backup IMproved） 1.1.1](https://www.halo.run/store/apps/app-dHakX)
  - [lightgallery.js 灯箱 1.2.1](https://github.com/halo-sigs/plugin-lightgallery)
  - [KaTeX 2.1.0](https://github.com/halo-sigs/plugin-katex)
  - [highlight.js 代码高亮 1.2.2](https://github.com/halo-sigs/plugin-highlightjs)
  - [对象存储（Amazon S3 协议） 1.11.0](https://github.com/halo-dev/plugin-s3)
  - [Markdown / HTML 内容块 1.2.0](https://github.com/halo-dev/plugin-hybrid-edit-block)
  - [微信公众号发布 1.3.1](https://www.halo.run/store/apps/app-Dtuoa)
  - [应用市场 1.3.0](https://www.halo.run/store/apps/app-VYJbF)
  - [Sitemap 1.1.2](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件 1.6.2](https://github.com/halo-dev/plugin-search-widget)
  - [RSS 1.3.0](https://github.com/halo-dev/plugin-feed)

### 使用的哪种方式运行？

Docker Compose

### 发生了什么？

文章打开后报 404 资源不存在，Post not found，重新发布后可正常显示

环境已保留：https://www.anye.xyz/archives/1719207299137
`/apis/content.halo.run/v1alpha1/posts/`已保存

### 复现步骤

目前不能稳定复现，但复现操作基本都是曾经在后台点击过刷新缓存按钮。

### 相关日志输出

```shell
2024-08-11T21:51:31.809+08:00 ERROR 7 --- [or-http-epoll-4] a.w.r.e.AbstractErrorWebExceptionHandler : [3d17b76f-8753]  500 Server Error for HTTP GET "/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=https://github.com/Anyexyz/it-tools/fork"
 
 org.springframework.web.reactive.function.client.WebClientRequestException: connection timed out after 30000 ms: github.com/20.205.243.166:443
 	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:136) ~[spring-webflux-6.1.10.jar:6.1.10]
 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
 Error has been observed at the following site(s):
 	*__checkpoint ⇢ Request to GET https://github.com/Anyexyz/it-tools/fork [DefaultWebClient]
 	*__checkpoint ⇢ run.halo.app.security.InitializeRedirectionWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.app.security.LoginRequiredForAccessWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.app.security.device.DeviceSessionFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.app.sms.security.SmsLoginAuthenticator [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.cache.page.PageCacheWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ AuthorizationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ ExceptionTranslationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ LogoutWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ ServerRequestCacheWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ LogoutPageGeneratingWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ AnonymousAuthenticationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ TotpAuthenticationFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ ReactorContextWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ CsrfWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ CorsWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ HttpHeaderWriterWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.comment.widget.captcha.CommentCaptchaFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.app.sms.security.SmsLoginAuthenticator [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.oauth.Oauth2Authenticator [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.oauth.Oauth2AuthorizationRequestRedirectWebFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ cc.ryanc.staticpages.endpoint.RewriteOnNotFoundFilter [DefaultWebFilterChain]
 	*__checkpoint ⇢ run.halo.app.webfilter.AdditionalWebFilterChainProxy [DefaultWebFilterChain]
 	*__checkpoint ⇢ HTTP GET "/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=https://github.com/Anyexyz/it-tools/fork" [ExceptionHandlingWebHandler]
 Original Stack Trace:
 		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:136) ~[spring-webflux-6.1.10.jar:6.1.10]
 		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:205) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:229) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:279) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onError(FluxContextWrite.java:121) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.maybeOnError(FluxConcatMapNoPrefetch.java:327) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.onNext(FluxConcatMapNoPrefetch.java:212) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.SinkManyEmitterProcessor.drain(SinkManyEmitterProcessor.java:476) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.SinkManyEmitterProcessor.tryEmitNext(SinkManyEmitterProcessor.java:273) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.SinkManySerialized.tryEmitNext(SinkManySerialized.java:100) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:194) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:205) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.netty.http.client.HttpClientConnect$MonoHttpConnect$ClientTransportSubscriber.onError(HttpClientConnect.java:311) ~[reactor-netty-http-1.1.20.jar:1.1.20]
 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:205) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onError(DefaultPooledConnectionProvider.java:172) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onError(FluxContextWrite.java:121) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.netty.internal.shaded.reactor.pool.AbstractPool$Borrower.fail(AbstractPool.java:492) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at reactor.netty.internal.shaded.reactor.pool.SimpleDequePool.lambda$drainLoop$9(SimpleDequePool.java:436) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:186) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:205) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.onError(DefaultPooledConnectionProvider.java:583) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:241) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:315) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.Operators.error(Operators.java:198) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.MonoError.subscribe(MonoError.java:53) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.Mono.subscribe(Mono.java:4568) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.6.7.jar:3.6.7]
 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.tryFailure(TransportConnector.java:576) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at reactor.netty.transport.TransportConnector$MonoChannelPromise.setFailure(TransportConnector.java:522) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at reactor.netty.transport.TransportConnector.lambda$doConnect$7(TransportConnector.java:261) ~[reactor-netty-core-1.1.20.jar:1.1.20]
 		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:583) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:559) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:118) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:616) ~[netty-transport-classes-epoll-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408) ~[netty-transport-classes-epoll-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 		at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
 Caused by: io.netty.channel.ConnectTimeoutException: connection timed out after 30000 ms: github.com/20.205.243.166:443
 	at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe$2.run(AbstractEpollChannel.java:615) ~[netty-transport-classes-epoll-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:153) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408) ~[netty-transport-classes-epoll-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:994) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.111.Final.jar:4.1.111.Final]
 	at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
 
 2024-08-11T22:03:00.148+08:00  WARN 7 --- [ndedElastic-428] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-428][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-11T22:03:00.149+08:00  WARN 7 --- [ndedElastic-428] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-428][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-11T22:03:00.197+08:00  WARN 7 --- [ndedElastic-428] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-428][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-11T23:18:52.125+08:00  WARN 7 --- [ndedElastic-436] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-436][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-11T23:18:52.126+08:00  WARN 7 --- [ndedElastic-436] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-436][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-11T23:18:52.171+08:00  WARN 7 --- [ndedElastic-436] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-436][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T01:23:52.802+08:00  WARN 7 --- [ndedElastic-445] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-445][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T01:23:52.807+08:00  WARN 7 --- [ndedElastic-445] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-445][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T01:23:52.852+08:00  WARN 7 --- [ndedElastic-445] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-445][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T03:00:00.001+08:00  INFO 7 --- [   scheduling-1] r.h.a.s.a.r.RememberTokenCleaner         : Expired remember me tokens have been cleaned up.
 2024-08-12T09:10:26.411+08:00 ERROR 7 --- [tor-tcp-epoll-2] run.jiewen.tool.auth.LywqPluginsUtil     : Error during authorization check: 请求失败！
 2024-08-12T09:10:30.213+08:00  WARN 7 --- [ndedElastic-465] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-465][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T09:10:30.223+08:00  WARN 7 --- [ndedElastic-465] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-465][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T09:10:30.282+08:00  WARN 7 --- [ndedElastic-465] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-465][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T11:38:49.098+08:00  WARN 7 --- [ndedElastic-475] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-475][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T11:38:49.099+08:00  WARN 7 --- [ndedElastic-475] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-475][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T11:38:49.143+08:00  WARN 7 --- [ndedElastic-475] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-475][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T15:01:05.773+08:00  WARN 7 --- [ndedElastic-487] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-487][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T15:01:05.773+08:00  WARN 7 --- [ndedElastic-487] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-487][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T15:01:05.823+08:00  WARN 7 --- [ndedElastic-487] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-487][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T16:32:05.692+08:00  WARN 7 --- [ndedElastic-499] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-499][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T16:32:05.693+08:00  WARN 7 --- [ndedElastic-499] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-499][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T16:32:05.743+08:00  WARN 7 --- [ndedElastic-499] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-499][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T17:22:10.978+08:00  WARN 7 --- [tor-tcp-epoll-3] s.b.a.a.m.OnlyOnceLoggingDenyMeterFilter : Reached the maximum number of URI tags for 'http.server.requests'.
 2024-08-12T17:22:12.835+08:00 ERROR 7 --- [tor-tcp-epoll-3] run.jiewen.tool.auth.LywqPluginsUtil     : Error during authorization check: 请求失败！
 2024-08-12T17:22:23.178+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Deleting old plugin file file:///root/.halo2/plugins/PluginPhotos-1.3.0.jar for plugin PluginPhotos, and new load location is file:///root/.halo2/plugins/PluginPhotos-1.4.0.jar.
 2024-08-12T17:22:23.179+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Deleted old plugin file file:///root/.halo2/plugins/PluginPhotos-1.3.0.jar for plugin PluginPhotos.
 2024-08-12T17:22:23.182+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Stop plugin 'PluginPhotos@1.3.0'
 2024-08-12T17:22:23.182+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Before publishing plugin stopping event for plugin PluginPhotos
 2024-08-12T17:22:23.185+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.controller.DefaultController     : Disposing controller run.halo.app.plugin.DefaultReactiveSettingFetcher
 2024-08-12T17:22:23.185+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.DefaultController     : Controller worker run.halo.app.plugin.DefaultReactiveSettingFetcher-worker-1 interrupted
 2024-08-12T17:22:23.185+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.DefaultController     : Controller worker run.halo.app.plugin.DefaultReactiveSettingFetcher-worker-1 is stopped
 2024-08-12T17:22:23.186+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.controller.DefaultController     : Controller run.halo.app.plugin.DefaultReactiveSettingFetcher is disposed
 2024-08-12T17:22:23.990+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : After publishing plugin stopping event for plugin PluginPhotos
 2024-08-12T17:22:23.991+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Stopping run.halo.photos.PhotoPlugin@e67fd20c for plugin PluginPhotos
 2024-08-12T17:22:23.995+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Stopped run.halo.photos.PhotoPlugin@e67fd20c for plugin PluginPhotos
 2024-08-12T17:22:23.995+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Closing plugin context for plugin PluginPhotos
 2024-08-12T17:22:24.000+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Closed plugin context for plugin PluginPhotos
 2024-08-12T17:22:24.001+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Reset plugin context for plugin PluginPhotos
 2024-08-12T17:22:24.001+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Unload plugin 'PluginPhotos@1.3.0'
 2024-08-12T17:22:24.012+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Loading plugin PluginPhotos from file:///root/.halo2/plugins/PluginPhotos-1.4.0.jar
 2024-08-12T17:22:24.018+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Plugin 'PluginPhotos@1.4.0' resolved
 2024-08-12T17:22:24.019+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Loaded plugin PluginPhotos from file:///root/.halo2/plugins/PluginPhotos-1.4.0.jar
 2024-08-12T17:22:24.019+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initializing setting and config map for plugin PluginPhotos
 2024-08-12T17:22:24.029+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initialized setting plugin-photos-settings for plugin PluginPhotos
 2024-08-12T17:22:24.032+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initialized config map plugin-photos-configmap for plugin PluginPhotos
 2024-08-12T17:22:24.037+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving logo resource for plugin PluginPhotos
 2024-08-12T17:22:24.037+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving main.js and style.css for plugin PluginPhotos
 2024-08-12T17:22:24.038+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Starting plugin PluginPhotos
 2024-08-12T17:22:24.039+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Start plugin 'PluginPhotos@1.4.0'
 2024-08-12T17:22:24.040+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Preparing starting plugin PluginPhotos
 2024-08-12T17:22:24.163+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Application context run.halo.app.plugin.PluginApplicationContext@e69780d4, started on Mon Aug 12 17:22:24 CST 2024, parent: org.springframework.context.support.GenericApplicationContext@35973862 for plugin PluginPhotos is created
 2024-08-12T17:22:24.171+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Before publishing plugin starting event for plugin PluginPhotos
 2024-08-12T17:22:24.173+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : After publishing plugin starting event for plugin PluginPhotos
 2024-08-12T17:22:24.174+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Starting run.halo.photos.PhotoPlugin@b4e23e45 for plugin PluginPhotos
 2024-08-12T17:22:24.249+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.index.IndexerFactoryImpl         : Start building index for type: /registry/core.halo.run/photos, please wait...
 2024-08-12T17:22:24.250+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.index.IndexerFactoryImpl         : Index for type: /registry/core.halo.run/photos built successfully, cost 1 ms
 2024-08-12T17:22:24.278+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.index.IndexerFactoryImpl         : Start building index for type: /registry/core.halo.run/photogroups, please wait...
 2024-08-12T17:22:24.279+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.index.IndexerFactoryImpl         : Index for type: /registry/core.halo.run/photogroups built successfully, cost 1 ms
 2024-08-12T17:22:24.280+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Started run.halo.photos.PhotoPlugin@b4e23e45 for plugin PluginPhotos
 2024-08-12T17:22:24.280+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Before publishing plugin started event for plugin PluginPhotos
 2024-08-12T17:22:24.280+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.controller.DefaultController     : Starting controller run.halo.app.plugin.DefaultReactiveSettingFetcher
 2024-08-12T17:22:24.282+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : After publishing plugin started event for plugin PluginPhotos
 2024-08-12T17:22:24.282+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.DefaultController     : Controller worker run.halo.app.plugin.DefaultReactiveSettingFetcher-worker-1 started
 2024-08-12T17:22:24.283+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.RequestSynchronizer   : Starting request(v1alpha1/ConfigMap) synchronizer...
 2024-08-12T17:22:24.283+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.RequestSynchronizer   : Started request(v1alpha1/ConfigMap) synchronizer.
 2024-08-12T17:22:24.283+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Started plugin PluginPhotos
 2024-08-12T17:22:24.305+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initializing setting and config map for plugin PluginPhotos
 2024-08-12T17:22:24.314+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initialized setting plugin-photos-settings for plugin PluginPhotos
 2024-08-12T17:22:24.328+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initialized config map plugin-photos-configmap for plugin PluginPhotos
 2024-08-12T17:22:24.335+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving logo resource for plugin PluginPhotos
 2024-08-12T17:22:24.336+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving main.js and style.css for plugin PluginPhotos
 2024-08-12T17:22:24.337+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Starting plugin PluginPhotos
 2024-08-12T17:22:24.337+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Started plugin PluginPhotos
 2024-08-12T17:22:30.035+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Deleting old plugin file file:///root/.halo2/plugins/PluginSearchWidget-1.6.1.jar for plugin PluginSearchWidget, and new load location is file:///root/.halo2/plugins/PluginSearchWidget-1.6.2.jar.
 2024-08-12T17:22:30.036+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Deleted old plugin file file:///root/.halo2/plugins/PluginSearchWidget-1.6.1.jar for plugin PluginSearchWidget.
 2024-08-12T17:22:30.036+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Stop plugin 'PluginSearchWidget@1.6.1'
 2024-08-12T17:22:30.036+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Before publishing plugin stopping event for plugin PluginSearchWidget
 2024-08-12T17:22:30.037+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.controller.DefaultController     : Disposing controller run.halo.app.plugin.DefaultReactiveSettingFetcher
 2024-08-12T17:22:30.037+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.DefaultController     : Controller worker run.halo.app.plugin.DefaultReactiveSettingFetcher-worker-1 interrupted
 2024-08-12T17:22:30.037+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.DefaultController     : Controller worker run.halo.app.plugin.DefaultReactiveSettingFetcher-worker-1 is stopped
 2024-08-12T17:22:30.038+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.controller.DefaultController     : Controller run.halo.app.plugin.DefaultReactiveSettingFetcher is disposed
 2024-08-12T17:22:30.812+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : After publishing plugin stopping event for plugin PluginSearchWidget
 2024-08-12T17:22:30.812+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Stopping run.halo.search.widget.SearchWidgetPlugin@13fabefe for plugin PluginSearchWidget
 2024-08-12T17:22:30.812+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Stopped run.halo.search.widget.SearchWidgetPlugin@13fabefe for plugin PluginSearchWidget
 2024-08-12T17:22:30.812+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Closing plugin context for plugin PluginSearchWidget
 2024-08-12T17:22:30.815+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Closed plugin context for plugin PluginSearchWidget
 2024-08-12T17:22:30.815+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Reset plugin context for plugin PluginSearchWidget
 2024-08-12T17:22:30.815+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Unload plugin 'PluginSearchWidget@1.6.1'
 2024-08-12T17:22:30.827+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Loading plugin PluginSearchWidget from file:///root/.halo2/plugins/PluginSearchWidget-1.6.2.jar
 2024-08-12T17:22:30.832+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Plugin 'PluginSearchWidget@1.6.2' resolved
 2024-08-12T17:22:30.833+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Loaded plugin PluginSearchWidget from file:///root/.halo2/plugins/PluginSearchWidget-1.6.2.jar
 2024-08-12T17:22:30.833+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initializing setting and config map for plugin PluginSearchWidget
 2024-08-12T17:22:30.839+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving logo resource for plugin PluginSearchWidget
 2024-08-12T17:22:30.839+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving main.js and style.css for plugin PluginSearchWidget
 2024-08-12T17:22:30.840+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Starting plugin PluginSearchWidget
 2024-08-12T17:22:30.840+08:00  INFO 7 --- [nReconciler-t-1] org.pf4j.AbstractPluginManager           : Start plugin 'PluginSearchWidget@1.6.2'
 2024-08-12T17:22:30.841+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Preparing starting plugin PluginSearchWidget
 2024-08-12T17:22:30.862+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Application context run.halo.app.plugin.PluginApplicationContext@1888732b, started on Mon Aug 12 17:22:30 CST 2024, parent: org.springframework.context.support.GenericApplicationContext@35973862 for plugin PluginSearchWidget is created
 2024-08-12T17:22:30.862+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Before publishing plugin starting event for plugin PluginSearchWidget
 2024-08-12T17:22:30.863+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : After publishing plugin starting event for plugin PluginSearchWidget
 2024-08-12T17:22:30.863+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Starting run.halo.search.widget.SearchWidgetPlugin@e0a42073 for plugin PluginSearchWidget
 2024-08-12T17:22:30.863+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Started run.halo.search.widget.SearchWidgetPlugin@e0a42073 for plugin PluginSearchWidget
 2024-08-12T17:22:30.863+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : Before publishing plugin started event for plugin PluginSearchWidget
 2024-08-12T17:22:30.863+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.e.controller.DefaultController     : Starting controller run.halo.app.plugin.DefaultReactiveSettingFetcher
 2024-08-12T17:22:30.865+08:00  INFO 7 --- [nReconciler-t-1] run.halo.app.plugin.SpringPlugin         : After publishing plugin started event for plugin PluginSearchWidget
 2024-08-12T17:22:30.866+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Started plugin PluginSearchWidget
 2024-08-12T17:22:30.866+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.DefaultController     : Controller worker run.halo.app.plugin.DefaultReactiveSettingFetcher-worker-1 started
 2024-08-12T17:22:30.867+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.RequestSynchronizer   : Starting request(v1alpha1/ConfigMap) synchronizer...
 2024-08-12T17:22:30.867+08:00  INFO 7 --- [tingFetcher-t-1] r.h.a.e.controller.RequestSynchronizer   : Started request(v1alpha1/ConfigMap) synchronizer.
 2024-08-12T17:22:30.898+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Initializing setting and config map for plugin PluginSearchWidget
 2024-08-12T17:22:30.917+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving logo resource for plugin PluginSearchWidget
 2024-08-12T17:22:30.918+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Resolving main.js and style.css for plugin PluginSearchWidget
 2024-08-12T17:22:30.919+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Starting plugin PluginSearchWidget
 2024-08-12T17:22:30.920+08:00  INFO 7 --- [nReconciler-t-1] r.h.a.c.e.reconciler.PluginReconciler    : Started plugin PluginSearchWidget
 2024-08-12T17:52:29.716+08:00  WARN 7 --- [ndedElastic-507] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-507][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T17:52:29.716+08:00  WARN 7 --- [ndedElastic-507] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-507][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T17:52:29.760+08:00  WARN 7 --- [ndedElastic-507] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-507][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T20:04:58.351+08:00  WARN 7 --- [ndedElastic-509] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-509][index] Deprecated unwrapped fragment expression "modules/sidebar.html" found in template index, line 19, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/sidebar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T20:04:58.352+08:00  WARN 7 --- [ndedElastic-509] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-509][index] Deprecated unwrapped fragment expression "modules/navbar.html" found in template index, line 23, col 12. Please use the complete syntax of fragment expressions instead ("~{modules/navbar.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
 2024-08-12T20:04:58.395+08:00  WARN 7 --- [ndedElastic-509] actStandardFragmentInsertionTagProcessor : [THYMELEAF][boundedElastic-509][index] Deprecated unwrapped fragment expression "modules/footer.html" found in template index, line 72, col 9. Please use the complete syntax of fragment expressions instead ("~{modules/footer.html}"). The old, unwrapped syntax for fragment expressions will be removed in future versions of Thymeleaf.
```


### 附加信息

_No response_

--- a/api/src/main/java/run/halo/app/extension/index/KeyComparator.java
+++ b/api/src/main/java/run/halo/app/extension/index/KeyComparator.java
@@ -7,7 +7,7 @@ public class KeyComparator implements Comparator<String> {
     public static final KeyComparator INSTANCE = new KeyComparator();
 
     @Override
-    public int compare(@Nullable  String a, @Nullable String b) {
+    public int compare(@Nullable String a, @Nullable String b) {
         if (a == null && b == null) {
             return 0;
         } else if (a == null) {
@@ -17,31 +17,169 @@ public class KeyComparator implements Comparator<String> {
             // null less than everything
             return -1;
         }
+        return compareStrings(a, b);
+    }
 
+    private int compareStrings(String a, String b) {
         int i = 0;
         int j = 0;
         while (i < a.length() && j < b.length()) {
-            if (Character.isDigit(a.charAt(i)) && Character.isDigit(b.charAt(j))) {
-                // handle number part
-                int num1 = 0;
-                int num2 = 0;
-                while (i < a.length() && Character.isDigit(a.charAt(i))) {
-                    num1 = num1 * 10 + (a.charAt(i++) - '0');
+            char charA = a.charAt(i);
+            char charB = b.charAt(j);
+
+            if (Character.isDigit(charA) && Character.isDigit(charB)) {
+                // Both characters are digits, compare as numbers
+                int compareResult = compareNumbers(a, b, i, j);
+                if (compareResult != 0) {
+                    return compareResult;
                 }
-                while (j < b.length() && Character.isDigit(b.charAt(j))) {
-                    num2 = num2 * 10 + (b.charAt(j++) - '0');
-                }
-                if (num1 != num2) {
-                    return num1 - num2;
-                }
-            } else if (a.charAt(i) != b.charAt(j)) {
-                // handle non-number part
-                return a.charAt(i) - b.charAt(j);
-            } else {
+
+                // Move indices past the compared number segments
+                i = moveIndexToNextNonDigit(a, i);
+                j = moveIndexToNextNonDigit(b, j);
+            } else if (charA == charB) {
+                // Characters are the same, continue
                 i++;
                 j++;
+            } else if (Character.isDigit(charA)) {
+                // If charA is digit and charB is not, digit comes first
+                return -1;
+            } else if (Character.isDigit(charB)) {
+                // If charB is digit and charA is not, digit comes first
+                return 1;
+            } else {
+                // Both are non-digits, compare directly
+                return Character.compare(charA, charB);
             }
         }
-        return a.length() - b.length();
+
+        return Integer.compare(a.length(), b.length());
+    }
+
+    private int compareNumbers(String a, String b, int startA, int startB) {
+        int i = startA;
+        int j = startB;
+
+        // Skip leading zeros for both numbers
+        while (i < a.length() && a.charAt(i) == '0') {
+            i++;
+        }
+        while (j < b.length() && b.charAt(j) == '0') {
+            j++;
+        }
+
+        // Compare lengths of remaining digits
+        int lengthA = countDigits(a, i);
+        int lengthB = countDigits(b, j);
+        if (lengthA != lengthB) {
+            return Integer.compare(lengthA, lengthB);
+        }
+
+        // Compare digits one by one
+        for (int k = 0; k < lengthA && i < a.length() && j < b.length(); k++, i++, j++) {
+            char charA = a.charAt(i);
+            char charB = b.charAt(j);
+            if (charA != charB) {
+                return Character.compare(charA, charB);
+            }
+        }
+
+        // If both numbers have decimal points, compare decimal parts
+        boolean hasDecimalA = i < a.length() && a.charAt(i) == '.';
+        boolean hasDecimalB = j < b.length() && b.charAt(j) == '.';
+        if (hasDecimalA || hasDecimalB) {
+            return compareDecimalNumbers(a, b, i, j);
+        }
+
+        return 0;
+    }
+
+    private int compareDecimalNumbers(String a, String b, int startA, int startB) {
+        // Find decimal point positions
+        int pointA = a.indexOf('.', startA);
+        int pointB = b.indexOf('.', startB);
+
+        // Compare integer parts before the decimal point
+        int integerComparison = compareIntegerPart(a, b, startA, startB, pointA, pointB);
+        if (integerComparison != 0) {
+            return integerComparison;
+        }
+
+        // Compare fractional parts after the decimal point
+        return compareFractionalPart(a, b, pointA + 1, pointB + 1);
+    }
+
+    private int compareIntegerPart(String a, String b, int startA, int startB, int pointA,
+        int pointB) {
+        int i = startA;
+        int j = startB;
+
+        while (i < pointA && a.charAt(i) == '0') {
+            i++;
+        }
+        while (j < pointB && b.charAt(j) == '0') {
+            j++;
+        }
+
+        int lengthA = pointA - i;
+        int lengthB = pointB - j;
+        if (lengthA != lengthB) {
+            return Integer.compare(lengthA, lengthB);
+        }
+
+        while (i < pointA && j < pointB) {
+            char charA = a.charAt(i);
+            char charB = b.charAt(j);
+            if (charA != charB) {
+                return Character.compare(charA, charB);
+            }
+            i++;
+            j++;
+        }
+
+        return 0;
+    }
+
+    private int compareFractionalPart(String a, String b, int i, int j) {
+        while (i < a.length() && j < b.length()
+            && Character.isDigit(a.charAt(i)) && Character.isDigit(b.charAt(j))) {
+            if (a.charAt(i) != b.charAt(j)) {
+                return Character.compare(a.charAt(i), b.charAt(j));
+            }
+            i++;
+            j++;
+        }
+
+        while (i < a.length() && Character.isDigit(a.charAt(i))) {
+            if (a.charAt(i) != '0') {
+                return 1;
+            }
+            i++;
+        }
+        while (j < b.length() && Character.isDigit(b.charAt(j))) {
+            if (b.charAt(j) != '0') {
+                return -1;
+            }
+            j++;
+        }
+
+        return 0;
+    }
+
+    private int countDigits(String s, int start) {
+        int count = 0;
+        while (start < s.length() && Character.isDigit(s.charAt(start))) {
+            count++;
+            start++;
+        }
+        return count;
+    }
+
+    private int moveIndexToNextNonDigit(String s, int index) {
+        while (index < s.length() && (Character.isDigit(s.charAt(index))
+            || s.charAt(index) == '.')) {
+            index++;
+        }
+        return index;
     }
 }

--- a/api/src/test/java/run/halo/app/extension/index/KeyComparatorTest.java
+++ b/api/src/test/java/run/halo/app/extension/index/KeyComparatorTest.java
@@ -4,6 +4,11 @@ import static org.assertj.core.api.Assertions.assertThat;
 
 import java.util.Arrays;
 import java.util.Comparator;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.TreeSet;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.RepeatedTest;
 import org.junit.jupiter.api.Test;
 
 /**
@@ -13,10 +18,10 @@ import org.junit.jupiter.api.Test;
  * @since 2.12.0
  */
 class KeyComparatorTest {
+    private final KeyComparator comparator = KeyComparator.INSTANCE;
 
     @Test
     void keyComparator() {
-        var comparator = KeyComparator.INSTANCE;
         String[] strings = {"103", "101", "102", "1011", "1013", "1021", "1022", "1012", "1023"};
         Arrays.sort(strings, comparator);
         assertThat(strings).isEqualTo(
@@ -34,7 +39,6 @@ class KeyComparatorTest {
 
     @Test
     void keyComparator2() {
-        var comparator = KeyComparator.INSTANCE;
         String[] strings =
             {"moment-101", "moment-102", "moment-103", "moment-1011", "moment-1013", "moment-1021",
                 "moment-1022", "moment-1012", "moment-1023"};
@@ -75,5 +79,419 @@ class KeyComparatorTest {
         strings = new String[] {null, "abc", "123", "xyz"};
         Arrays.sort(strings, comparator);
         assertThat(strings).isEqualTo(new String[] {"123", "abc", "xyz", null});
+    }
+
+    @Test
+    void complexStringTest() {
+        var strings = new String[] {
+            "1719560085223",
+            "1719564195757",
+            "AJHQ9JKT",
+            "1719565849173",
+            "5InykKCe",
+            "123123123",
+            "adJhTqEo",
+            "123123",
+            "Ahvcq7Wn",
+            "asda",
+            "b5jHcxfe"
+        };
+        Arrays.sort(strings, comparator);
+        assertThat(strings).containsExactly(
+            "5InykKCe",
+            "123123",
+            "123123123",
+            "1719560085223",
+            "1719564195757",
+            "1719565849173",
+            "AJHQ9JKT",
+            "Ahvcq7Wn",
+            "adJhTqEo",
+            "asda",
+            "b5jHcxfe"
+        );
+    }
+
+    @Test
+    void complexButSkewedStringTest() {
+        var strings = new String[] {
+            "chu-shi-hua-gong-neng-you-hua-halo-2.9.0-fa-bu",
+            "cxcc",
+            "d",
+            "dddd",
+            "ddddd",
+            "de-dao",
+            "dENMr6tX",
+            "dian-shang-ke-fu",
+            "dong-tai-she-ji-shi-xi-25jie",
+            "eeeeeeee",
+            "ejqRrTp4",
+            "Fh8Jd09T",
+            "g5gZaGvS",
+        };
+        Arrays.sort(strings, comparator);
+        assertThat(strings).containsExactly(
+            "Fh8Jd09T",
+            "chu-shi-hua-gong-neng-you-hua-halo-2.9.0-fa-bu",
+            "cxcc",
+            "d",
+            "dENMr6tX",
+            "dddd",
+            "ddddd",
+            "de-dao",
+            "dian-shang-ke-fu",
+            "dong-tai-she-ji-shi-xi-25jie",
+            "eeeeeeee",
+            "ejqRrTp4",
+            "g5gZaGvS"
+        );
+    }
+
+    @Test
+    void mixLetterCaseStringTest() {
+        var strings = new String[] {
+            "VpLBxBJ7", "AJHQ9JKT", "asda", "Tq5EgH2V", "Fh8Jd09T", "J7KMLQeK", "adJhTqEo",
+            "Ahvcq7Wn",
+        };
+        Arrays.sort(strings, comparator);
+        assertThat(strings).containsExactly(
+            "AJHQ9JKT", "Ahvcq7Wn", "Fh8Jd09T", "J7KMLQeK", "Tq5EgH2V", "VpLBxBJ7", "adJhTqEo",
+            "asda"
+        );
+    }
+
+    @Test
+    void mixLetterCaseAndNumberTest() {
+        var strings = new String[] {
+            "1719565849173", "1719564195757", "1703040584263",
+            "AJHQ9JKT", "Ahvcq7Wn", "Fh8Jd09T", "adJhTqEo",
+            "asda", "1703053590063", "1702955288482",
+            "zhi-chi-bei-fen-hui-fu-halo-2.8.0-fa-bu",
+            "zhi-chi-ge-ren-zhong-xin-halo-2.11.0-fa-bu",
+            "J7KMLQeK", "Tq5EgH2V", "VpLBxBJ7",
+            "b5jHcxfe", "cao-ni-ma-a-huang-jian-ming", "chu-ji-ying-jian-kai-fa",
+            "ddddd", "de-dao", "dian-shang-ke-fu", "eeeeeeee", "ejqRrTp4",
+            "halo-maintainer-2023-nian-du-bang-dan", "hello-halo", "hello-world",
+            "dong-tai-she-ji-shi-xi-25jie", "halo-nuan-dong-li-yu-quan-chang-qi-zhe-qi", "hello",
+            "kai-fang-gong-gong-api-halo-2.5.0-fa-bu",
+            "xing-neng-you-hua-yu-gong-neng-gai-jin-halo-2.13-fa-bu", "ye-wu-tuo-zhan-jing-li",
+            "ying-qu-jing-mei-zhou-bian-halo-ying-yong-shi-chang-zhu-ti-you-jiang-zheng-ji",
+            "zhi-chi-bao-chi-deng-lu-hui-hua-halo-2.16.0-fa-bu",
+        };
+        Arrays.sort(strings, comparator);
+        assertThat(strings).containsExactly(
+            "1702955288482",
+            "1703040584263",
+            "1703053590063",
+            "1719564195757",
+            "1719565849173",
+            "AJHQ9JKT",
+            "Ahvcq7Wn",
+            "Fh8Jd09T",
+            "J7KMLQeK",
+            "Tq5EgH2V",
+            "VpLBxBJ7",
+            "adJhTqEo",
+            "asda",
+            "b5jHcxfe",
+            "cao-ni-ma-a-huang-jian-ming",
+            "chu-ji-ying-jian-kai-fa",
+            "ddddd",
+            "de-dao",
+            "dian-shang-ke-fu",
+            "dong-tai-she-ji-shi-xi-25jie",
+            "eeeeeeee",
+            "ejqRrTp4",
+            "halo-maintainer-2023-nian-du-bang-dan",
+            "halo-nuan-dong-li-yu-quan-chang-qi-zhe-qi",
+            "hello",
+            "hello-halo",
+            "hello-world",
+            "kai-fang-gong-gong-api-halo-2.5.0-fa-bu",
+            "xing-neng-you-hua-yu-gong-neng-gai-jin-halo-2.13-fa-bu",
+            "ye-wu-tuo-zhan-jing-li",
+            "ying-qu-jing-mei-zhou-bian-halo-ying-yong-shi-chang-zhu-ti-you-jiang-zheng-ji",
+            "zhi-chi-bao-chi-deng-lu-hui-hua-halo-2.16.0-fa-bu",
+            "zhi-chi-bei-fen-hui-fu-halo-2.8.0-fa-bu",
+            "zhi-chi-ge-ren-zhong-xin-halo-2.11.0-fa-bu");
+    }
+
+    @Test
+    public void sortingWithComplexStringsTest() {
+        List<String> strings = Arrays.asList("abc10", "abc2", "abc1", "abc20", "abc100");
+        strings.sort(comparator);
+        assertThat(strings).containsExactly("abc1", "abc2", "abc10", "abc20", "abc100");
+    }
+
+    @Test
+    public void sortingWithDecimalStringsTest() {
+        List<String> strings =
+            Arrays.asList("1.2", "1.10", "1.1", "1.20", "1.02", "1.22", "1.001", "1.002");
+        strings.sort(comparator);
+        assertThat(strings).containsExactly("1.001", "1.002", "1.02", "1.1", "1.10", "1.2", "1.20",
+            "1.22");
+    }
+
+    @Test
+    public void treeSetWithComparatorTest() {
+        TreeSet<String> set = new TreeSet<>(comparator);
+        set.add("abc123");
+        set.add("abc1");
+        set.add("abc12");
+        set.add("abc2");
+
+        assertThat(set).containsExactly("abc1", "abc2", "abc12", "abc123");
+    }
+
+    @Test
+    public void testTreeMap_WithComparator() {
+        TreeMap<String, String> map = new TreeMap<>(comparator);
+        map.put("2024-08-29", "date1");
+        map.put("2024-08-28", "date2");
+        map.put("2024-08-30", "date3");
+
+        assertThat(map.keySet()).containsExactly("2024-08-28", "2024-08-29", "2024-08-30");
+        assertThat(map.get("2024-08-29")).isEqualTo("date1");
+    }
+
+    @Test
+    public void integerPartDifferentTest() {
+        // Create strings with different integer parts to cover the compareIntegerPart code
+        // block
+        String[] strings = {"abc10", "abc2", "abc1", "abc20", "abc10022229"};
+        Arrays.sort(strings, comparator);
+
+        String[] expectedOrder = {"abc1", "abc2", "abc10", "abc20", "abc10022229"};
+        assertThat(strings).containsExactly(expectedOrder);
+    }
+
+    @Test
+    public void integerPartDifferentWithDecimalTest() {
+        // To specifically test integer part comparison
+        String str1 = "abc12.5";
+        String str2 = "abc11.5";
+
+        // Compare should return a positive number since "12" > "11"
+        assertThat(comparator.compare(str1, str2)).isPositive();
+
+        String str3 = "abc11.5";
+        String str4 = "abc12.5";
+
+        // Compare should return a negative number since "11" < "12"
+        assertThat(comparator.compare(str3, str4)).isNegative();
+
+        // Test for multiple decimal points
+        assertThat(comparator.compare("1.23.4", "1.23")).isGreaterThan(0);
+        assertThat(comparator.compare("1.23", "1.23.4")).isLessThan(0);
+
+        assertThat(comparator.compare("1..23", "1.23")).isLessThan(0);
+        assertThat(comparator.compare("1.23..", "1.23")).isGreaterThan(0);
+
+        assertThat(comparator.compare("", "1.23")).isLessThan(0);
+        assertThat(comparator.compare("1.23", "")).isGreaterThan(0);
+
+        assertThat(comparator.compare("1.23", "1.23")).isZero();
+    }
+
+    @Nested
+    class ComparatorCharacteristicTest {
+        @Test
+        public void reflexiveTest() {
+            // Reflexive: a == a should always return 0
+            assertThat(comparator.compare("test", "test")).isZero();
+            assertThat(comparator.compare("", "")).isZero();
+            assertThat(comparator.compare("123", "123")).isZero();
+            assertThat(comparator.compare(null, null)).isZero();
+        }
+
+        @Test
+        public void symmetricTest() {
+            // Symmetric: a > b implies b < a
+            assertThat(comparator.compare("123", "test")).isNegative();
+            assertThat(comparator.compare("test", "123")).isPositive();
+
+            assertThat(comparator.compare("1.023", "1.23")).isNegative();
+            assertThat(comparator.compare("1.23", "1.023")).isPositive();
+        }
+
+        @Test
+        public void transitiveTest() {
+            // Transitive: a > b and b > c implies a > c
+            assertThat(comparator.compare("test2", "test1")).isPositive();
+            assertThat(comparator.compare("test1", "test0")).isPositive();
+            assertThat(comparator.compare("test2", "test0")).isPositive();
+        }
+
+        @RepeatedTest(50)
+        public void consistencyTest() {
+            // Consistency: a == b should always return 0 if not changed
+            assertThat(comparator.compare("123abc", "123abc")).isZero();
+            assertThat(comparator.compare("test", "test")).isZero();
+            assertThat(comparator.compare("123abc", "123abc"))
+                .isEqualTo(comparator.compare("123abc", "123abc"));
+        }
+
+        @Test
+        public void withNumbersTest() {
+            // Numbers should be compared numerically
+            assertThat(comparator.compare("item2", "item10")).isNegative();
+            assertThat(comparator.compare("item10", "item2")).isPositive();
+            assertThat(comparator.compare("item10", "item10")).isZero();
+        }
+
+        @Test
+        public void mixedContentTest() {
+            // Mixed content comparison
+            assertThat(comparator.compare("abc123", "abc124")).isNegative();
+            assertThat(comparator.compare("abc124", "abc123")).isPositive();
+            assertThat(comparator.compare("abc123", "abc123")).isZero();
+        }
+
+        @Test
+        public void nullHandlingTest() {
+            // Null handling
+            assertThat(comparator.compare(null, "test")).isPositive();
+            assertThat(comparator.compare("test", null)).isNegative();
+            assertThat(comparator.compare(null, null)).isZero();
+        }
+
+        @Test
+        public void lengthDifferenceTest() {
+            // Length difference should affect comparison
+            assertThat(comparator.compare("test", "testa")).isNegative();
+            assertThat(comparator.compare("testa", "test")).isPositive();
+        }
+
+        @Test
+        public void specialCharactersTest() {
+            // Special character comparison
+            assertThat(comparator.compare("a#1", "a#2")).isNegative();
+            assertThat(comparator.compare("a#2", "a#1")).isPositive();
+            assertThat(comparator.compare("a#1", "a#1")).isZero();
+        }
+
+        @Test
+        public void emptyStringsTest() {
+            // Empty string comparison
+            assertThat(comparator.compare("", "test")).isNegative();
+            assertThat(comparator.compare("test", "")).isPositive();
+            assertThat(comparator.compare("", "")).isZero();
+        }
+    }
+
+    @Nested
+    class ComparatorEdgeTest {
+        @Test
+        public void pureNumbersTest() {
+            assertThat(comparator.compare("123", "123")).isEqualTo(0);
+            assertThat(comparator.compare("123", "124")).isLessThan(0);
+            assertThat(comparator.compare("124", "123")).isGreaterThan(0);
+            // Leading zeros
+            assertThat(comparator.compare("00123", "123") > 0).isTrue();
+        }
+
+        @Test
+        public void mumbersWithOverflowTest() {
+            // Max long value
+            String largeNumber1 = "9223372036854775807";
+            // One more than max long value
+            String largeNumber2 = "9223372036854775808";
+            assertThat(comparator.compare(largeNumber1, largeNumber2)).isLessThan(0);
+            assertThat(comparator.compare(largeNumber2, largeNumber1)).isGreaterThan(0);
+
+            // large number str comparison
+            assertThat(comparator.compare("123456789012345678901234567890",
+                "123456789012345678901234567891")).isLessThan(0);
+            assertThat(comparator.compare("123456789012345678901234567890",
+                "123456789012345678901234567891")).isNotPositive();
+            assertThat(comparator.compare("999999999999999999999999999999",
+                "999999999999999999999999999998")).isGreaterThan(0);
+            assertThat(comparator.compare("999999999999999999999999999999",
+                "999999999999999999999999999998")).isNotNegative();
+            assertThat(comparator.compare("9999999999999999999999999999999999999999999999",
+                "9999999999999999999999999999999999999999999998")).isGreaterThan(0);
+            assertThat(comparator.compare("100000000000000000000000000000",
+                "100000000000000000000000000000")).isEqualTo(0);
+
+            // This specific case is to test the overflow for a real-world scenario
+            assertThat(comparator.compare("5InykKCe", "1710683457874") < 0).isTrue();
+            assertThat(comparator.compare("5InykKce", "1717477435943") > 0).isFalse();
+
+            assertThat(comparator.compare("0",
+                "9999999999999999999999999999999999999999999998")).isLessThan(0);
+            assertThat(comparator.compare("9999999999999999999999999999999999999999999998",
+                "0")).isGreaterThan(0);
+        }
+
+        @Test
+        public void decimalStringsTest() {
+            assertThat(comparator.compare("123.45", "123.45")).isEqualTo(0);
+            assertThat(comparator.compare("123.45", "123.46")).isLessThan(0);
+            assertThat(comparator.compare("123.46", "123.45")).isGreaterThan(0);
+            // Decimal equivalence
+            assertThat(comparator.compare("123.5", "123.50")).isLessThan(0);
+        }
+
+        @Test
+        public void lettersAndNumbersTest() {
+            assertThat(comparator.compare("abc123", "abc123")).isEqualTo(0);
+            assertThat(comparator.compare("abc123", "abc124")).isLessThan(0);
+            assertThat(comparator.compare("abc124", "abc123")).isGreaterThan(0);
+            assertThat(comparator.compare("abc123", "abcd123")).isLessThan(0);
+        }
+
+        @Test
+        public void pureLettersTest() {
+            assertThat(comparator.compare("abc", "abc")).isEqualTo(0);
+            assertThat(comparator.compare("abc", "abcd")).isLessThan(0);
+            assertThat(comparator.compare("abcd", "abc")).isGreaterThan(0);
+            // Case sensitivity
+            assertThat(comparator.compare("ABC", "abc")).isLessThan(0);
+        }
+
+        @Test
+        public void dateStringsTest() {
+            assertThat(comparator.compare("2024-08-29", "2024-08-29")).isEqualTo(0);
+            assertThat(comparator.compare("2024-08-29", "2024-08-30")).isLessThan(0);
+            assertThat(comparator.compare("2024-08-30", "2024-08-29")).isGreaterThan(0);
+
+            // Time comparison
+            assertThat(comparator.compare("2024-08-29T12:00:00.000Z", "2024-08-29T12:00:00.001Z"))
+                .isLessThan(0);
+            assertThat(comparator.compare("2024-08-29T12:00:00.001Z", "2024-08-29T12:00:00.000Z"))
+                .isGreaterThan(0);
+            assertThat(comparator.compare("2024-08-29T12:00:00.000Z", "2024-08-29T12:00:01.000Z"))
+                .isLessThan(0);
+            assertThat(comparator.compare("2024-08-29T12:00:01.000Z", "2024-08-29T12:00:00.000Z"))
+                .isGreaterThan(0);
+            assertThat(comparator.compare("2024-08-29T12:00:00.000Z", "2024-08-29T12:01:00.000Z"))
+                .isLessThan(0);
+            assertThat(comparator.compare("2024-08-29T12:01:00.000Z", "2024-08-29T12:00:00.000Z"))
+                .isGreaterThan(0);
+            assertThat(comparator.compare("2024-08-29T12:00:00.000Z", "2024-08-29T13:00:00.000Z"))
+                .isLessThan(0);
+            assertThat(comparator.compare("2024-08-29T13:00:00.000Z", "2024-08-29T12:00:00.000Z"))
+                .isGreaterThan(0);
+            assertThat(comparator.compare("2024-08-29T12:00:00.000Z", "2024-08-30T12:00:00.000Z"))
+                .isLessThan(0);
+            assertThat(comparator.compare("2024-08-30T12:00:00.000Z", "2024-08-29T12:00:00.000Z"))
+                .isGreaterThan(0);
+        }
+
+        @Test
+        public void booleanStringsTest() {
+            assertThat(comparator.compare("true", "false")).isGreaterThan(0);
+            assertThat(comparator.compare("false", "true")).isLessThan(0);
+            assertThat(comparator.compare("true", "true")).isEqualTo(0);
+            assertThat(comparator.compare("false", "false")).isEqualTo(0);
+        }
+
+        @Test
+        public void complexMixedStringsTest() {
+            assertThat(comparator.compare("abc123xyz456", "abc123xyz456")).isEqualTo(0);
+            assertThat(comparator.compare("abc123xyz456", "abc124xyz456")).isLessThan(0);
+            assertThat(comparator.compare("abc124xyz456", "abc123xyz456")).isGreaterThan(0);
+            assertThat(comparator.compare("abc123xyz456", "abc123xyz457")).isLessThan(0);
+        }
     }
 }

